src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        FigureTypes figureType = getFigureType();
+
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Color color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case RECTANGLE:
+                int sideA = new Random().nextInt(100) + 1;
+                int sideB = new Random().nextInt(100) + 1;

```

Comment:
let's also create Random outside the switch/case and use this object to create random values